### PR TITLE
FAT-1031: Given an item Id, a user Id, and a pickup location, attempt to create a hold request when the applicable request policy disallows holds

### PR DESCRIPTION
**Pull request by yusuf-murodov-epam**
## Purpose
[FAT-1031](https://issues.folio.org/browse/FAT-1031)
_Given an item Id, a user Id, and a pickup location, attempt to create a hold request when the applicable request policy disallows holds_

## Approach
- Reproduced test scenario with UI.
- Learned necessary sent requests and then necessary received responses 
- Based on the request/response created karate test scenario

## Learning
- learn karate framework from original doc file on the github
- looked other scenarios how they are described.

### CODE REVIEW

File: mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature [hunk 1]
```diff
@@ -45,3 +46,33 @@ Feature: Requests tests
...
     Then status 422
     And match $.errors[0].message == 'Page requests are not allowed for this patron and item combination'
+
+  Scenario: Given an item Id, a user Id, and a pickup location, attempt to create a hold request when the applicable request policy disallows holds
+    * def userBarcode = 'FAT-1031UBC'
+    * def itemBarcode = 'FAT-1031IBC'
+    * def extRequestTypesForRequestPolicy = ["Page", "Recall"]
+
+    # post a material type and policies
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostPolicies') { extRequestTypes: #(extRequestTypesForRequestPolicy) }
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature:56]: there are 5 different policies in that method and it isn't really clear for which of them variable "extRequestTypes" are being sent. So I guess let's rename it into extRequestTypesFor...Policy

File: mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature [hunk 2]
```diff
@@ -45,3 +46,33 @@ Feature: Requests tests
...
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostLocation')
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostHoldings')
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostItem') { extItemId: #(itemId), extItemBarcode: #(itemBarcode) }
+
+    # post a group and an user
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostGroup')
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostUser') { extUserId: #(userId), extUserBarcode: #(userBarcode) }
+
+    # post a request and verify that the user are not allowed to create a page request
+    * def requestId = call uuid1
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature:70]: Could we move it at the beginning of the test ? Please don't say me later I selected bunch of them. Just one line

Comment (yusuf-murodov-epam) [reply mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature]: ok, fixed

File: mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature (hunk 1 above)
Comment (yusuf-murodov-epam) [reply mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature]: fixed

**Review by viktor-draban — APPROVED** [review]

**Pull request merged**